### PR TITLE
Scheduler implementation

### PR DESCRIPTION
**Pull request by mathialo**
Proposal for a new scheduler. Inspired by tha apscheduler, but with two important changes: Respecting a cancellation token both in the main loop and in sleeps, and allowing tasks to be triggered manually while still respecting the no overlap rule.

### CODE REVIEW

File: cognite/extractorutils/unstable/scheduling/_schedules.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    def next(self) -> int:
+        return int(self._cron.get_next(start_time=time()))
+
+
+class IntervalSchedule(Schedule):
+    def __init__(self, interval: int) -> None:
+        self._interval = interval
+        self._jitter = randint(0, interval)
+
+        self._next = int(time() + self._jitter)
```
Comment (einarmo) [inline cognite/extractorutils/unstable/scheduling/_schedules.py:27]: I think jitter needs a rethink as discussed.

**Review by einarmo — COMMENTED** [review]
Should maybe have some logging, and perhaps a test.

File: cognite/extractorutils/unstable/scheduling/_schedules.py (hunk 1 above)
Comment (mathialo) [reply cognite/extractorutils/unstable/scheduling/_schedules.py]: Agree, I'll take it out for now and we can consider it later.

**Review by einarmo — APPROVED** [review]
Rocket Emoji

**Pull request merged**